build.gradle.kts

```diff
@@ -225,12 +227,27 @@ tasks {
 	}
 
 	javadoc {
-		// of all the javadoc checks (accessibility, html, missing, reference, syntax; see
-		// https://docs.oracle.com/javase/8/docs/technotes/tools/unix/javadoc.html#BEJEFABE)
-		// disable the warning for missing comments and tags because they spam the output
-		// (it does often not make sense to comment every tag; e.g. the @return tag on annotations)
-		(options as CoreJavadocOptions).addStringOption("Xdoclint:accessibility,html,syntax,reference", "-quiet")
-		options.encoding = "UTF-8"
+		// Exclude internal implementation package from javadoc
+		exclude("org/junitpioneer/internal")
+
+		options {
+			// Cast to standard doclet options, see https://github.com/gradle/gradle/issues/7038#issuecomment-448294937
+			this as StandardJavadocDocletOptions
+
+			links = listOf("https://junit.org/junit5/docs/current/api/")
+
+			// Set javadoc `--release` flag (affects which warnings and errors are reported)
+			// (Note: Gradle adds one leading '-' to the option on its own)
+			addStringOption("-release", targetJavaVersion.getMajorVersion())
+
+			// Enable doclint, but ignore warnings for missing tags, see
+			// https://docs.oracle.com/en/java/javase/17/docs/specs/man/javadoc.html#additional-options-provided-by-the-standard-doclet
+			// The Gradle option methods are rather misleading, but a boolean `true` value just makes sure the flag
+			// is passed to javadoc, see https://github.com/gradle/gradle/issues/2354
+			addBooleanOption("Xdoclint:all,-missing", true)
+			encoding = "UTF-8"
```

Comment:
I would move this up just before `links`. Then, the `options` block becomes:

```kotlin
this as StandardJavadocDocletOptions

encoding = "UTF-8"
links = listOf("https://junit.org/junit5/docs/current/api/")

// options
// ...
```

IMHO a bit easier to read, what do you think?